src/resources/Resource.ts

```diff
@@ -102,43 +102,48 @@ export abstract class Resource extends EventEmitter {
         acceptType?: string,
         keyFormatter = toSnakeCase
     ): DefinedRoute {
-        const api: RestAPI = this.api;
-
-        return function route<A, B>(
+        return <A, B>(
             originalData?: SendData<A>,
             callback?: ResponseCallback<B>,
             auth?: AuthParams,
             pathParams: Record<string, string> = {}
-        ): Promise<B> {
+        ): Promise<B> => {
             /**
-             * Sanitises and ensures that the data is recreated as an object with default prototypes.
+             * Sanitizes and ensures that the data is recreated as an object with default prototypes.
              * In rare cases if the data was created with `Object.create(null)` (no prototypes),
              * this will cause the data to be sent as `WebKitFormBoundary` instead of regular JSON object.
              *
              * Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create#custom_and_null_objects
              */
             const data = getObjectType(originalData) === "object" ? { ...originalData } : originalData;
 
-            const url: string = compilePath(path, pathParams);
+            const url = compilePath(path, pathParams);
 
-            const missingPathParams: string[] = (url.match(/:([a-z]+)/gi) || []).map((m: string) => m.replace(":", ""));
-            const missingParams: string[] = missingKeys(data, required);
```

Comment:
This could be made more efficient by doing `obj ? required.find(key => data[key] === undefined) : keys[0]`.
Since we are only interested in the first missing param

Reply:
Yeah! I could remove the other function, nice catch!